### PR TITLE
Makes healthbot tx gas limit 10% greater than estimated

### PR DESCRIPTION
**Pull request by brunocalza**
# Summary

Suddenly, the `healthbot` transactions started to revert. And the reason for that is that the gas limit was being exceeded. This PR fixes that by making the `healthbot` tx gas limit 10% greater than the estimated gas limit.

# Implementation overview

**The important part of implementation:**
- Refactors the Tableland Ethereum client to make a call to `EstimateGas` and bumps that value by 10%
- Makes the multiplier configurable

Additional changes:
- It bumps the `abi` gen binary to the latest version
- It updates the Tableland Ethereum `abi` and `bytecode`
- Refactors the toolkit to use the Tableland Ethereum client.

### CODE REVIEW

File: pkg/tables/impl/ethereum/client.go [hunk 1]
```diff
@@ -137,6 +137,7 @@ func (c *Client) RunSQL(
 			From:      auth.From,
 			Nonce:     big.NewInt(0).SetInt64(nonce),
 			GasTipCap: gasTipCap,
+			GasLimit:  100000,
```
Comment (joewagner) [inline pkg/tables/impl/ethereum/client.go:140]: @brunocalza I'm sure on this, but you might be able to use alchemy to do the gas estimate for you?
https://docs.alchemy.com/reference/eth-estimategas

File: Makefile [hunk 2]
```diff
@@ -10,24 +10,25 @@ HTTP_PORT ?= 8080
 GO_BINDATA=go run github.com/go-bindata/go-bindata/v3/go-bindata@v3.1.3
 SQLC=go run github.com/kyleconroy/sqlc/cmd/sqlc@v1.15.0
 GOVVV=go run github.com/ahmetb/govvv@v0.3.0 
+ABIGEN=go run github.com/ethereum/go-ethereum/cmd/abigen@v1.11.4
```
Comment (brunocalza) [inline Makefile:13]: Using the latest version of `abigen` now

File: cmd/healthbot/config.go [hunk 3]
```diff
@@ -39,6 +39,7 @@ type ChainConfig struct {
 		GatewayEndpoint             string
 		ContractAddr                string
 		SuggestedGasPriceMultiplier float64
+		EstimatedGasLimitMultiplier float64
```
Comment (brunocalza) [inline cmd/healthbot/config.go:42]: Added new `EstimatedGasLimitMultiplier` config to the healthbot

File: cmd/healthbot/counterprobe/counterprobe.go [hunk 4]
```diff
@@ -142,7 +145,8 @@ func (cp *CounterProbe) increaseCounterValue(ctx context.Context) error {
 	txnHash, err := cp.client.Write(
 		ctx,
 		fmt.Sprintf("update %s set counter=counter+1", cp.tableName),
-		clientV1.WithSuggestedPriceMultiplier(cp.suggGasPriceMultiplier))
+		clientV1.WithSuggestedPriceMultiplier(cp.suggGasPriceMultiplier),
+		clientV1.WithEstimatedGasLimitMultiplier(cp.estGasLimitMultiplier))
```
Comment (brunocalza) [inline cmd/healthbot/counterprobe/counterprobe.go:149]: passing the config value to the write call of our Go client

File: cmd/healthbot/counterprobe/counterprobe_test.go [hunk 5]
```diff
@@ -22,7 +22,7 @@ func TestProduction(t *testing.T) {
 	client, err := clientV1.NewClient(ctx, wallet, clientV1.NewClientChain(chain))
 	require.NoError(t, err)
 
-	cp, err := New("optimism-mainnet", client, "Runbook_24", time.Second, time.Second*10, 1)
+	cp, err := New("optimism-mainnet", client, "Runbook_24", time.Second, time.Second*10, 1, 1)
```
Comment (brunocalza) [inline cmd/healthbot/counterprobe/counterprobe_test.go:25]: adjusting the test

File: cmd/toolkit/smart_contract.go [hunk 6]
```diff
@@ -6,13 +6,14 @@ import (
 	"fmt"
 	"math/big"
```
Comment (brunocalza) [inline cmd/toolkit/smart_contract.go:8]: This file implements the contract calls of our toolkit CLI. The changes here are just to use the Tables Ethereum Client

File: docker/deployed/testnet/healthbot/config.json [hunk 7]
```diff
@@ -27,7 +27,8 @@
                 "Tablename": "${HEALTHBOT_ETHEREUM_GOERLI_TABLE}"
             },
             "OverrideClient": {
-                "SuggestedGasPriceMultiplier": 1.2
+                "SuggestedGasPriceMultiplier": 1.2,
+                "EstimatedGasLimitMultiplier": 1.1
```
Comment (brunocalza) [inline docker/deployed/testnet/healthbot/config.json:31]: Healthbot configured to multiply gas limit by 10%

**Review by joewagner — APPROVED** [review]
Looks good to me!

File: Makefile (hunk 2 above)
Comment (avichalp) [reply Makefile:13]: question: do we need to ensure that the same version of `abigen` is used in `evm-tableland` while deploying the `tableland-tables` contract?

File: pkg/tables/impl/ethereum/client.go [hunk 8]
```diff
@@ -127,6 +132,25 @@ func (c *Client) RunSQL(
...
+	if err != nil {
+		return nil, fmt.Errorf("parsing abi: %s", err)
+	}
+
+	data, err := tablesABI.Pack("runSQL", []interface{}{addr, table.ToBigInt(), statement}...)
+	if err != nil {
+		return nil, fmt.Errorf("abi packing: %s", err)
+	}
+
+	gas, err := c.backend.EstimateGas(ctx, ethereum.CallMsg{
```
Comment (avichalp) [inline pkg/tables/impl/ethereum/client.go:145]: 👍

File: Makefile (hunk 2 above)
Comment (brunocalza) [reply Makefile:13]: Well! The `abigen` binary is just converting the `abi.json` to Go code. I don't think the `abigen` version is too relevant. What is relevant is to have the `abi.json` and `bytecode.bin` compatible with what is deployed.

Comment (avichalp) [reply Makefile:13]: Got it. Make sense!

**Pull request merged**